modal/experimental/__init__.py

```diff
@@ -88,6 +89,53 @@ async def list_deployed_apps(environment_name: str = "", client: Optional[_Clien
     return app_infos
 
 
+@synchronizer.create_blocking
+async def get_app_objects(
+    app_name: str, *, environment_name: Optional[str] = None, client: Optional[_Client] = None
+) -> dict[str, Union[_Function, _Cls]]:
+    """Experimental interface for retrieving a dictionary of the Functions / Clses in an App.
+
+    The return value is a dictionary mapping names to Function or Cls objects.
+
+    We plan to support this functionality through a stable API in the future. It's likely that
+    the stable API will look different (it will probably be a method on the App object itself).
+    Additionally, note that the returned objects are currently not hydrated, but a future stable
+    API would likely return hydrated objects instead.
```

Comment:
I guess we could run the handles through parallel explicit `.hydrate()` calls here , at the cost of another (longest in the set) round trip in this function (currently there are already 2 so it will be rather slow).

Reply:
Actually! It doesn't really make sense to pre-hydrate a Cls, since you don't know the parameters in this context. Bah.